src/Ocean/OceanBC.jl

```diff
@@ -86,3 +87,89 @@ Prescribe the net inward temperature flux across the boundary by `flux`,
 a function with signature `flux(problem, state, aux, t)`, returning the flux (in m⋅K/s).
 """
 struct TemperatureFlux <: TemperatureBC end
+
+"""
+    ocean_boundary_state!(nf, boundaries::Tuple, ::HBModel,
+                          Q⁺, A⁺, n, Q⁻, A⁻, bctype)
+applies boundary conditions for the first-order and gradient fluxes
+dispatches to a function in OceanBoundaryConditions.jl based on bytype defined by a problem such as SimpleBoxProblem.jl
+"""
+@generated function ocean_boundary_state!(
```

Comment:
This is starting to get fancy

Reply:
the `@generated` magic is thanks to @simonbyrne, I've just stolen the idea